kafka-fetch-rewrite/src/lib.rs

```diff
@@ -6,24 +6,39 @@ use kafka_protocol::records::{
 };
 use serde::{Deserialize, Serialize};
 use shotover::frame::kafka::{KafkaFrame, ResponseBody};
-use shotover::frame::Frame;
+use shotover::frame::{Frame, MessageType};
 use shotover::message::Messages;
-use shotover::transforms::{Transform, TransformBuilder, TransformConfig, Transforms, Wrapper};
+use shotover::transforms::{
+    DownChainProtocol, Transform, TransformBuilder, TransformConfig, TransformContextBuilder,
+    TransformContextConfig, UpChainProtocol, Wrapper,
+};
 
 #[derive(Deserialize, Serialize, Debug, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct KafkaFetchRewriteConfig {
     pub result: String,
 }
 
+const NAME: &str = "KafkaFetchRewrite";
 #[typetag::serde(name = "KafkaFetchRewrite")]
 #[async_trait(?Send)]
 impl TransformConfig for KafkaFetchRewriteConfig {
-    async fn get_builder(&self, _chain_name: String) -> Result<Box<dyn TransformBuilder>> {
+    async fn get_builder(
+        &self,
+        _transform_context: TransformContextConfig,
+    ) -> Result<Box<dyn TransformBuilder>> {
         Ok(Box::new(KafkaFetchRewriteBuilder {
             result: self.result.clone(),
         }))
     }
+
+    fn up_chain_protocol(&self) -> UpChainProtocol {
```

Comment:
What are `up_chain_protocol` and `down_chain_protocol` used for?

Reply:
I went to the link to the documentation, and then realized I never documented. Fixing now...

Reply:
https://github.com/shotover/shotover-proxy/pull/1699 I added missing documentation here.
The main point is that they define which protocols the transform supports so that shotover can detect a misconfiguration and give a useful error at startup instead of confusing runtime behavior when a transform receives a message it is not capable of handling.
They were added as mandatory methods in shotover 0.4, hence why they are added in this PR.